dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,265 @@
+                         onlyFiles,  sortBy,  sortByDesc,  excludeLinks,  languageId, false);
+    }
+    
+	/**
+	 * Gets the Folders, HTMLPages, Links, FileAssets under the specified folderId.
+	 *
+	 * @param user
+	 * @param folderId
+	 * @param offset
+	 * @param maxResults
+	 * @param filter
+	 * @param mimeTypes
+	 * @param extensions
+	 *
+	 * @param showWorking   If true, returns the working version of HTMLPages, Links and FileAssets in the folder.
+	 * 						If false, returns the live version of HTMLPages, Links and FileAssets in the folder.
+	 *
+	 * @param showArchived  If true, includes the archived version of HTMLPages, Links and FileAssets in the folder.
+	 *
+	 * @param noFolders
+	 * @param onlyFiles
+	 * @param sortBy
+	 * @param sortByDesc
+	 * @param excludeLinks
+	 * @return
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+    @Deprecated
+	public default Map<String, Object> getFolderContent(final User user, final String folderId,
+                                                final int offset, final int maxResults, final String filter, final List<String> mimeTypes,
+                                                final List<String> extensions, final boolean showWorking, final boolean showArchived, final boolean noFolders,
+                                                final boolean onlyFiles, final String sortBy, final boolean sortByDesc, final boolean excludeLinks, final long languageId, final boolean dotAssets)
+			throws DotSecurityException, DotDataException {
+
+	    final boolean showPages =! onlyFiles;
+
+	    final BrowserQuery browserQuery = BrowserQuery.builder()
+	                    .showDotAssets(dotAssets)
+	                    .showLinks(!excludeLinks)
+	                    .showExtensions(extensions)
+	                    .withFilter(filter)
+	                    .withHostOrFolderId(folderId)
+	                    .withLanguageId(languageId)
+	                    .offset(offset)
+	                    .showFiles(true)
+	                    .showPages(showPages)
+	                    .showFolders(!noFolders)
+	                    .showArchived(showArchived)
+	                    .showWorking(showWorking)
+                        .showMimeTypes(mimeTypes)
+                        .maxResults(maxResults)
+	                    .sortBy(sortBy)
+	                    .sortByDesc(sortByDesc)
+	                    .withUser(user).build();
+
+	    return getFolderContent(browserQuery);
+	}
+
+    /**
+     * Returns a map with two elements:
+     *     1 - "total": size of the total elements available
+     *     2 - "list":  collection of contentlets as a map to be showed in the browsers
+     * @param browserQuery {@link BrowserQuery}
+     * @return Map result
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'getFolderContent': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)